### PR TITLE
Implement multimatch pattern array matching

### PR DESCRIPTION
**Pull request by jmatsushita**
Using only minimatch it's sometimes quite hard to express certain patterns or impossible. Being able to use arrays of patterns with multimatch makes things much easier to reason about for me. Maybe for others too?

### CODE REVIEW

**Comment by Rodrigo-Ludgero** [comment]
Hi @lambtron, It would be possible to join this PR, to the repository, to enable the use of arrays of patterns?

Thanks in advance.

File: package.json [hunk 1]
```diff
@@ -9,11 +9,12 @@
     "debug": "~0.7.4",
     "extend": "~1.2.1",
     "minimatch": "^0.2.14",
```
Comment (axe312ger) [inline package.json]: sure we also need minimatch?

File: lib/index.js [hunk 2]
```diff
@@ -181,7 +182,7 @@ function matcher(cols){
 
     for (var key in matchers){
       var m = matchers[key];
-      if (m && m.match(file)) matches.push(key);
```
Comment (axe312ger) [inline lib/index.js]: .length is enough

File: lib/index.js [hunk 3]
```diff
@@ -162,12 +162,13 @@ function matcher(cols){
   keys.forEach(function(key){
     var opts = cols[key];
     if (!opts.pattern) return;
-    matchers[key] = new Matcher(opts.pattern);
+
+    matchers[key] = { match: function(file) {
```
Comment (axe312ger) [inline lib/index.js]: i vote for clean up this intendation

**Pull request closed without merging**